api/src/test/java/org/pmiops/workbench/cohorts/CohortMaterializationServiceTest.java

```diff
@@ -0,0 +1,225 @@
+
+    Cohort cohort = new Cohort();
+    cohort.setWorkspaceId(workspace.getWorkspaceId());
+    cohort.setName("males");
+    cohort.setType("AOU");
+    Gson gson = new Gson();
+    cohort.setCriteria(gson.toJson(SearchRequests.males()));
+    cohortDao.save(cohort);
+
+    Cohort cohort2 = new Cohort();
+    cohort2.setWorkspaceId(workspace.getWorkspaceId());
+    cohort2.setName("all genders");
+    cohort2.setType("AOU");
+    cohort2.setCriteria(gson.toJson(SearchRequests.allGenders()));
+    cohortDao.save(cohort2);
+
+    cohortReview = new CohortReview();
+    cohortReview.setCdrVersionId(cdrVersion.getCdrVersionId());
+    cohortReview.setCohortId(cohort2.getCohortId());
+    cohortReview.setMatchedParticipantCount(3);
+    cohortReview.setReviewedCount(2);
+    cohortReview.setReviewSize(3);
+    cohortReviewDao.save(cohortReview);
+
+    participantCohortStatusDao.save(makeStatus(cohortReview.getCohortReviewId(), 1L, CohortStatus.INCLUDED));
+    participantCohortStatusDao.save(makeStatus(cohortReview.getCohortReviewId(), 2L, CohortStatus.EXCLUDED));
+
+    ConceptService conceptService = new ConceptService(entityManager, conceptDao, conceptSynonymDao);
+
+    this.cohortMaterializationService = new CohortMaterializationService(fieldSetQueryBuilder,
+        annotationQueryBuilder, participantCohortStatusDao, cdrBigQuerySchemaConfigService, conceptService);
+
+  }
+
+  @Test
+  public void testMaterializeAnnotationQueryNoPagination() {
+    FieldSet fieldSet = new FieldSet();
+    fieldSet.setAnnotationQuery(new AnnotationQuery());
+    MaterializeCohortResponse response =
+        cohortMaterializationService.materializeCohort(cohortReview, SearchRequests.allGenders(),
+            null, 0, makeRequest(fieldSet, 1000));
+    ImmutableMap<String, Object> p1Map = ImmutableMap.of("person_id", 1L, "review_status", "INCLUDED");
+    assertResults(response, p1Map);
+    assertThat(response.getNextPageToken()).isNull();
+  }
+
+  @Test
+  public void testMaterializeAnnotationQueryWithPagination() {
+    FieldSet fieldSet = new FieldSet();
+    fieldSet.setAnnotationQuery(new AnnotationQuery());
+    MaterializeCohortRequest request = makeRequest(fieldSet, 1);
+    request.setStatusFilter(ImmutableList.of(CohortStatus.INCLUDED, CohortStatus.EXCLUDED));
+    MaterializeCohortResponse response =
+        cohortMaterializationService.materializeCohort(cohortReview, SearchRequests.allGenders(), null, 0, request);
+    ImmutableMap<String, Object> p1Map = ImmutableMap.of("person_id", 1L, "review_status", "INCLUDED");
+    assertResults(response, p1Map);
+    assertThat(response.getNextPageToken()).isNotNull();
+
+    request.setPageToken(response.getNextPageToken());
+    MaterializeCohortResponse response2 =
+        cohortMaterializationService.materializeCohort(cohortReview, SearchRequests.allGenders(), null, 0, request);
+    ImmutableMap<String, Object> p2Map = ImmutableMap.of("person_id", 2L, "review_status", "EXCLUDED");
```

Comment:
I think I'm missing something - is this test hitting BigQuery? If not - how is it working?

Reply:
Annotations are stored in Cloud SQL. So it's not hitting BigQuery here.